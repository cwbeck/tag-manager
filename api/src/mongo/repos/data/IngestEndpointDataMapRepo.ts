import { injectable } from 'inversify';
import IngestEndpointDataMap from '../../models/data/IngestEndpointDataMap';
import UnderIngestEndpointRevisionControl from './abstractions/UnderIngestEndpointRevisionControl';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class IngestEndpointDataMapRepo extends UnderIngestEndpointRevisionControl<IngestEndpointDataMap> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                ___persisting_id: 1,
                _revision_id: 1,
            },
            unique: true,
        },
    ];
}
