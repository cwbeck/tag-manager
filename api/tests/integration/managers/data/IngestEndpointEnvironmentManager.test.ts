import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import IngestEndpointDataMapRepo from '../../../../src/mongo/repos/data/IngestEndpointDataMapRepo';
import Mock = jest.Mock;

describe('IngestEndpointEnvironmentManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches ingest endpoint environment', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();

        const mockSutModel = TestUtils.buildIngestEndpointEnvironment();
        const mockSutRepo = {} as IngestEndpointDataMapRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);

        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockReturnValue(mockSutRepo);

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getIngestEndpointEnvironment(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                    custom_domain
                    ingest_endpoint_revision {
                        id
                    }
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
