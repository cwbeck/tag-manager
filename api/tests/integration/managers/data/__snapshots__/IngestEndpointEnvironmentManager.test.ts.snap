// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngestEndpointEnvironmentManager fetches ingest endpoint environment 1`] = `
Object {
  "data": Object {
    "getIngestEndpointEnvironment": Object {
      "custom_domain": null,
      "id": "5edf4c48e9615d9a1c66baff",
      "ingest_endpoint_revision": Object {
        "id": "5edf4c48e9615d9a1c66baff",
      },
      "name": "test environment",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
